File: lib/models/destination.dart
import 'package:cloud_firestore/cloud_firestore.dart';
class Destination {
  final String id;
  final String name;
  final String description;
  final String location;
  final double rating;
  final List<String> images;
  final String category;
  final GeoPoint coordinates;
  final bool isFeatured;

  Destination({
    required this.id,
    required this.name,
    required this.description,
    required this.location,
    required this.rating,
    required this.images,
    required this.category,
    required this.coordinates,
    this.isFeatured = false,
  });

  factory Destination.fromFirestore(DocumentSnapshot doc) {
    Map<String, dynamic> data = doc.data() as Map<String, dynamic>;
    return Destination(
      id: doc.id,
      name: data['name'] ?? '',
      description: data['description'] ?? '',
      location: data['location'] ?? '',
      rating: (data['rating'] ?? 0).toDouble(),
      images: List<String>.from(data['images'] ?? []),
      category: data['category'] ?? '',
      coordinates: data['coordinates'] ?? const GeoPoint(0, 0),
      isFeatured: data['isFeatured'] ?? false,
    );
  }

  Map<String, dynamic> toFirestore() {
    return {
      'name': name,
      'description': description,
      'location': location,
      'rating': rating,
      'images': images,
      'category': category,
      'coordinates': coordinates,
      'isFeatured': isFeatured,
    };
  }
}

File: backend/models/Destination.js
const mongoose = require('mongoose');

const destinationSchema = new mongoose.Schema({
  name: {
    type: String,
    required: [true, 'Please provide a destination name'],
    trim: true,
    maxlength: [100, 'Destination name cannot be more than 100 characters']
  },
  description: {
    type: String,
    required: [true, 'Please provide a description']
  },
  location: {
    type: String,
    required: [true, 'Please provide a location']
  },
  coordinates: {
    latitude: {
      type: Number,
      required: true
    },
    longitude: {
      type: Number,
      required: true
    }
  },
  rating: {
    type: Number,
    default: 4.5,
    min: [1, 'Rating must be at least 1'],
    max: [5, 'Rating cannot be more than 5']
  },
  images: [{
    public_id: {
      type: String,
      required: true
    },
    url: {
      type: String,
      required: true
    }
  }],
  category: {
    type: String,
    enum: ['wildlife', 'adventure', 'culture', 'relaxation', 'landscape'],
    required: true
  },
  activities: [String],
  bestTimeToVisit: [String],
  entryFee: {
    local: Number,
    foreign: Number
  },
  isFeatured: {
    type: Boolean,
    default: false
  },
  createdAt: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model('Destination', destinationSchema);